### ISSUE TITLE
Testnet should be using the MempoolAdmitter

### ISSUE THREAD

**Issue by friedger** [blockstack/stacks-blockchain#1374]
When calling a sequence of two contract (`buy-non-expiring` and `has-valid-license` of OI License contract) calls twice (nonces: 1, 2, 1, 2), the testnet panicks

``` 
thread 'main' panicked at 'Error while processing block - InvalidStacksBlock("Block aa50242b262e3a515b46c5ae91a4d5c107005c90b08001f24a33f8da9c9fe267 state root mismatch: expected e8ed3930147dbfae7db566402c63b823834fa0c91334ae8ef56052278c5a5def, got 6421ae83d7b2052d4d191fe4c5244e04297be0f86c420205f0a7c6d55c24c864")', src/testnet/node.rs:311:31
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:77
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1052
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1426
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:204
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:224
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:472
  11: rust_begin_unwind
             at src/libstd/panicking.rs:380
  12: std::panicking::begin_panic_fmt
             at src/libstd/panicking.rs:334
  13: blockstack_lib::testnet::node::Node::process_tenure
             at src/testnet/node.rs:311
  14: blockstack_lib::testnet::run_loop::RunLoop::start
             at src/testnet/run_loop.rs:185
  15: blockstack_core::main
             at src/main.rs:215
  16: std::rt::lang_start::{{closure}}
             at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/rt.rs:67
  17: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:52
  18: std::panicking::try::do_call
             at src/libstd/panicking.rs:305
  19: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:86
  20: std::panicking::try
             at src/libstd/panicking.rs:281
  21: std::panic::catch_unwind
             at src/libstd/panic.rs:394
  22: std::rt::lang_start_internal
             at src/libstd/rt.rs:51
  23: std::rt::lang_start
             at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/rt.rs:67
  24: main
  25: __libc_start_main
  26: _start

```

**Comment by jcnelson** [comment]
Hey @friedger , thanks for reporting!

Can you post the transactions you submitted, to help us reproduce the issue?

Also, what branch is this on?  Have you seen @kantai's latest PR here? https://github.com/blockstack/stacks-blockchain/pull/1373

**Comment by friedger** [comment]
I was running from main branch.
The issue is reproducible by adding an extra parameter to a public function and call it with one less arguments (https://github.com/friedger/oi-license/commit/fce3aa9f1e09bc7a5a6493fac638881cdd9ea6a2)

I start the testnet like this:
``` 
cargo run --bin blockstack-core testnet  1> log.txt 2> log_debug.txt
``` 
Transaction 1 is created by 
```
cargo run --bin blockstack-cli generate-sk --testnet > keys.json

mempool=$(cat log.txt | head -2 | tail -1)
mempool=${mempool#*** Mempool path: }
echo $mempool
secretKey=$(cat keys.json | jq '.secretKey')
cargo run --bin blockstack-cli publish ${secretKey//\"} 0 0 oi-license ../../friedger/oi-license/contracts/license/license.clar --testnet | xxd -r -p > tx1.bin
cp tx1.bin $mempool

```

The script below for transaction 2 and 3 is run twice:
```
mempool=$(cat log.txt | head -2 | tail -1)
mempool=${mempool#*** Mempool path: }
echo $mempool
secretKey=$(cat keys.json | jq '.secretKey')
contractAddress=$(cat keys.json | jq '.stacksAddress')
cargo run --bin blockstack-cli contract-call ${secretKey//\"} 0 1 ${contractAddress//\"} oi-license buy-non-expiring --testnet | xxd -r -p > tx2.bin
cp tx2.bin $mempool

cargo run --bin blockstack-cli contract-call ${secretKey//\"} 0 2 ${contractAddress//\"} oi-license has-valid-license -e \'${contractAddress//\"} --testnet | xxd -r -p > tx3.bin
cp tx3.bin $mempool

```

**Comment by friedger** [comment]
I have submitted the bin files for transactions via https://blockstack.zendesk.com/agent/tickets/10066

**Comment by friedger** [comment]
The panic also occurs on feature/genesis

**Comment by lgalabru** [comment]
Hi @friedger, thank you for opening this issue, we will address this very soon!

**Comment by jcnelson** [comment]
I don't think the solution here is to use the `MempoolAdmitter`.  We want to find out why an invalid block can cause a panic.  Miners are not required to use the mempool to filter transactions -- they can publish arbitrary data for their blocks.

**Comment by lgalabru** [comment]
Yes, this issue is also referenced by #1376.

**Comment by jcnelson** [comment]
@friedger @lgalabru I think I have a fix: https://github.com/blockstack/stacks-blockchain/pull/1379

**Comment by diwakergupta** [comment]
Should be fixed by PR for #1218
